### PR TITLE
fix: habit completion and confetti positioning

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR
Fixed habit completion tracking and improved confetti animation positioning

### What changed?
- Added `setMockHabitCompletions` function to safely update mock habit completion data
- Fixed habit completion counter to properly initialize at 1 for new completions
- Adjusted confetti animation positioning and made it non-interactive
- Updated confetti container to prevent interference with underlying UI elements

### How to test?
1. Complete a habit for the first time - verify it starts at 1 completion
2. Complete habits multiple times - verify the counter increments correctly
3. Reach a habit goal - confirm confetti appears in the correct position
4. Verify you can still interact with UI elements when confetti is playing

### CODE REVIEW

**Comment by owengretzinger** [comment]
* **#139** <a href="https://app.graphite.dev/github/pr/habits-together/app/139?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/139?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — APPROVED** [review]
